terraform/sponsored_bucket.tf

```diff
@@ -158,3 +159,39 @@ data "aws_iam_policy_document" "sponsored_bucket" {
     ]
   }
 }
+
+module "sponsored_embargo_bucket" {
+  source              = "./modules/dandiset_bucket"
+  bucket_name         = "dandiarchive-embargo"
+  versioning          = false
+  log_bucket_name     = "dandiarchive-embargo-logs"
+  log_bucket_owner_id = data.aws_canonical_user_id.sponsored_account.id
+  policy_json         = data.aws_iam_policy_document.sponsored_embargo_bucket.json
+  providers = {
+    aws = aws.sponsored
+  }
+}
+
+data "aws_iam_policy_document" "sponsored_embargo_bucket" {
+  version = "2008-10-17"
+
+  statement {
+    sid = "dandi-api"
+    principals {
+      type        = "AWS"
+      identifiers = [data.aws_iam_user.api.arn]
+    }
+    actions = [
+      "s3:*",
+    ]
+    resources = [
+      "${module.sponsored_embargo_bucket.bucket_arn}",
+      "${module.sponsored_embargo_bucket.bucket_arn}/*",
+    ]
+    condition {
```

Comment:
This condition only applies to actions performed by `data.aws_iam_user.api.arn`. If another principal gets access to the bucket, it won't be enforced. Is that intended?

Reply:
No other principal should ever have access to the embargo buckets. Regardless, setting `BucketOwnerEnforced` should resolve that.

Reply:
BucketOwnerEnforced isn't an allowed keyword, I think you need to disable ACLs entirely. I'm going to leave this for later.